registrations/management/commands/fix_pmtct_registrations.py

```diff
@@ -0,0 +1,48 @@
+from django.core.management.base import BaseCommand
+
+from registrations.models import Registration
+from registrations.tasks import validate_subscribe
+
+
+class Command(BaseCommand):
+    help = ("Find and fixes all PMTCT registrations that aren't validated"
+            "because of the edd date not being populated")
+
+    def handle(self, *args, **kwargs):
+
+        registrations = Registration.objects.filter(
+            validated=False,
+            reg_type__in=("pmtct_prebirth", "pmtct_postbirth")).iterator()
+
+        updates = 0
+
+        for registration in registrations:
+
+            related_regs = Registration.objects.filter(
+                    validated=True,
+                    registrant_id=registration.registrant_id
+                ).exclude(reg_type__in=("pmtct_prebirth", "pmtct_postbirth")).\
+                order_by('-created_at')
+
+            updated = False
+            for related_reg in related_regs:
+                for field in ['edd', 'language', 'mom_dob', 'operator_id']:
+
+                    if (related_reg.data.get(field) and
+                            not registration.data.get(field)):
+                        registration.data[field] = related_reg.data[field]
+                        updated = True
```

Comment:
Wouldn't this mean that we're using the oldest registration we have for the field value? It also doesn't assure us that all the fields that are missing have been updated.

We could also let the database do this work. `related_regs.filter(data__has_key=field).first()` will return the most recent registration with that field, or `None` if a registration doesn't exist.

So a loop like `for field in set(('edd', ...)).difference(registration.data.keys())`, and in there we have the `related_regs.filter(data__has_key=field).first()`, and if any of the fields result in `None`, then we can not update the registration (since we don't have all the fields we need), and we skip that registration.